### ISSUE TITLE
Save.yml deleted

### ISSUE THREAD

**Issue by blablubbabcDEV** [Shopkeepers/Shopkeepers#284]
**Migrated from:** https://dev.bukkit.org/projects/shopkeepers/issues/281

**Originally posted by MasterMithrandir (Dec 27, 2014):** 

http://pastebin.com/ZCCegNcDUpdated Spigot to the last build, and found out something was wrong whith Shopkeepers. They had all dissapeared.I checked save.yml, everything was fine (maybe I didn't load the current file state), then I reloaded the plugin and I got the error in pastebin.I think this database loss should be no problem really, as I always have backups. I just thought I would mention this issue.Looks like there's some problem whith Books? All shopkeepers selling books have converted their book data into this:- !!net.minecraft.server.v1_8_R1.ChatComponentText {}

**Comment by blablubbabcDEV** [comment]
**Originally commented by blablubbabc (Dec 27, 2014):** 

Probably an issue with spigot. Could you check which item in your save.yml contains the 'net.minecraft.server.1_8_R1' tag? And also which exact version you were previously using?
Edit: did those books previously have this tag or only after updating spigot?
Anyways i assume that's something spigot has to fix.

            Edited Dec 27, 2014

**Comment by blablubbabcDEV** [comment]
**Originally commented by MasterMithrandir (Dec 27, 2014):** 


http://pastebin.com/Ma4LkGvy

I apply the old save.yml again, and reload various times, but this time the taggs do not appear.

Allthough I do get that error on log.

I'm going to keep reloading untill the tags appear whith the problematic items, I'm sure they will.

**Comment by blablubbabcDEV** [comment]
**Originally commented by MasterMithrandir (Dec 27, 2014):** 


One of the problematic items (I think they are mostly books).

Before:

http://pastebin.com/NHEEU0xe

After waiting a few minutes whith the plugin loaded, and then restarting:

http://pastebin.com/8wBBekAy

There's something very strange going on, I'm going to have to cut off the broken shopkeepers in order to get the plugin working.
            Edited Dec 27, 2014

**Comment by blablubbabcDEV** [comment]
**Originally commented by MasterMithrandir (Dec 27, 2014):** 

I managed to stabilize the database. But I'm sure that if someone tries to make a shop whith some books, or some weird item on it, it's going to break the whole plugin stability.

**Comment by blablubbabcDEV** [comment]
**Originally commented by blablubbabc (Dec 27, 2014):** 

You could try it again by updating to the latest spigot build. It should be fixed now again.

**Comment by blablubbabcDEV** [comment]
**Originally commented by MasterMithrandir (Dec 27, 2014):** 


Ok, now revising.

**Comment by blablubbabcDEV** [comment]
**Originally commented by MasterMithrandir (Dec 27, 2014):** 


http://pastebin.com/frxT1uCi

I tried the old config again, It's still giving errors.

Server version: Spigot-5ffe41c-c76552e (MC: 1.8)
            Edited Dec 27, 2014

**Comment by blablubbabcDEV** [comment]
**Originally commented by MasterMithrandir (Dec 27, 2014):** 

I get this:

http://pastebin.com/jejb5V9y

each time I reload.

The database seem's unperturbed (besides from the fact that some shops have been removed due to suposed inactivity), but shopkeepers seem unperturbed. The weird tags don't appear, at least for now. (They might come back on following restarts)

**Comment by blablubbabcDEV** [comment]
**Originally commented by MasterMithrandir (Dec 27, 2014):** 

I've also noticed Naming shopkeepers now makes you accidentaly talk on chat.

This was previously fixed, but has re-emerged

**Comment by blablubbabcDEV** [comment]
**Originally commented by blablubbabc (Dec 27, 2014):** 

[at]MasterMithrandir:
First problem: I have passed this issue forward to the people of spigot.
Second issue: The concurrent modification error is related to the remove-shopkeepers-on-inactivity feature. I can fix it once I am back at home, in a few days. For now you should deactivate this in the config again.
Third issue: I can't test this myself currently. Might help to know which chat plugin you are using, or any other plugin which offers chat related features. Could you also please try it on a fresh server with only shopkeepers running, to make sure it's not an issue with another plugin? Because that would be my current guess.

Best you create a separate ticket for this/those issue(s).

**Comment by blablubbabcDEV** [comment]
**Originally commented by blablubbabc (Dec 27, 2014):** 

[at]MasterMithrandir:
Okay, please try again with the latest spigot build.

**Comment by blablubbabcDEV** [comment]
**Originally commented by MasterMithrandir (Dec 27, 2014):** 


- Updating to Spigot's latest buld.

- Set "remove-shopkeepers-on-inactivity: 0"

- Opened two organized tickets on issues 2 and 3:

http://dev.bukkit.org/bukkit-plugins/shopkeepers/tickets/283-remove-shopkeepers-on-inactivity-error/

http://dev.bukkit.org/bukkit-plugins/shopkeepers/tickets/282-setting-shopkeeper-name-shows-on-chat/

**Comment by blablubbabcDEV** [comment]
**Originally closed by blablubbabc (Dec 27, 2014)**
